app/models/course.rb

```diff
@@ -157,8 +157,8 @@ def new_draft_attributes
     end
   end
 
-  scope :by_provider_name_ascending, -> { includes(:provider).order("provider.provider_name asc") }
-  scope :by_provider_name_descending, -> { includes(:provider).order("provider.provider_name desc") }
+  scope :ascending_canonical_order, -> { includes(:provider).order("provider.provider_name asc, name asc") }
```

Comment:
Ideally these scopes would reference a scope on `Provider` to avoid `Course` being coupled to the `Provider` DB schema. Also, they should use `joins` not `includes` as we don't need the `Provider` fields to be returned with the query.